test_runner/regress/test_timeline_archive.py

```diff
@@ -94,3 +94,29 @@ def test_timeline_archive(neon_simple_env: NeonEnv):
         timeline_id=parent_timeline_id,
         state=TimelineArchivalState.ARCHIVED,
     )
+
+    # Test that the leaf can't be unarchived
+    with pytest.raises(
+        PageserverApiException,
+        match="Cannot unarchive timeline which has archived ancestor",
+    ) as exc:
+        assert timeline_path.exists()
+
+        ps_http.timeline_archival_config(
+            tenant_id=env.initial_tenant,
+            timeline_id=leaf_timeline_id,
+            state=TimelineArchivalState.UNARCHIVED,
+        )
```

Comment:
Is there no allowed_error needed?

Reply:
This is logged at INFO level, allowed errors are only needed for things logged at ERROR level:

```
2024-08-28T12:41:23.850636Z  INFO request{method=POST path=/v1/tenant/01bd7b878dcdecfb6b942da698da4c83/timeline/c43a119e5402ecfe3555bd75c1f63133/archival_config request_id=6bb3f965-8852-4f88-ba5b-09b21ba10c7b}: Error processing HTTP request: Precondition failed: Cannot unarchive timeline which has archived ancestor: 1d69987f2e7bf1556e7eecdbf6a7782d
```